include/nlohmann/detail/input/input_adapters.hpp

```diff
@@ -61,7 +64,11 @@ class file_input_adapter
 
     std::char_traits<char>::int_type get_character() noexcept
     {
-        return std::fgetc(m_file);
+        if (JSON_HEDLEY_LIKELY(m_file != nullptr))
+        {
+            return std::fgetc(m_file);
+        }
+        return std::char_traits<char>::eof(); // LCOV_EXCL_LINE
```

Comment:
Are we not able to write tests that assert?

Reply:
I could try to set the `FILE` pointer to null after creating it.

Reply:
Wierd. :-)

Reply:
It would have to be a build without ubsan and with assertions disabled.

Reply:
I removed the `if`. The assertion, function annotation, and documentation state that the passed pointer must not be null. I don't see the point of playing what-if and spending more effort to simulate a situation where a user that misuses the API is still saved from the consequences.

Reply:
Okay.